megamek/src/megamek/client/ui/swing/BoardEditor.java

```diff
@@ -1622,47 +1554,69 @@ private void boardSaveImage(boolean ignoreUnits) {
     }
 
     /**
-     * Opens a file dialog box to select a file to save as; saves the board to
-     * the file.
+     * Saves the board to a .board file. 
+     * When saveAs is true, acts as a Save As... by opening a file chooser dialog.
+     * When saveAs is false, it will directly save to the current board file name,
+     * if it is known and otherwise act as Save As... 
      */
-    private boolean boardSaveAs() {
-        // First, correct connection issues
+    private boolean boardSave(boolean saveAs) {
+        // Correct connection issues and do a validation.
         correctExits();
-        JFileChooser fc = new JFileChooser("data" + File.separator + "boards");
+        validateBoard(false); 
+        
+        // Choose a board file to save to if this was
+        // called as "Save As..." or there is no current filename
+        if (curBoardFile == null || saveAs) {
+            if (!chooseSaveBoardFile()) {
+                return false;
+            }
+        }
+        butSourceFile.setEnabled(true);
+
+        // write the board
+        try {
+            OutputStream os = new FileOutputStream(curBoardFile);
```

Comment:
This should be moved into the try statement itself (`try (...)`) so if board.save throws, `os` will be cleaned up.

Reply:
Done.